### PR TITLE
profileing and tuning python apis

### PR DESCRIPTION
**Pull request by you-n-g**
before tuning: resNet on cifar-10 will spend about 330s every epoch with multiverso
after tuning: a epoch is reduced to about 110s every epoch with multiverso.

### CODE REVIEW

File: binding/python/multiverso/api.py [hunk 1]
```diff
@@ -45,6 +45,9 @@ def get(self, size):
     def add(self, data):
         raise NotImplementedError("You must implement the add method.")
```
Comment (huxuan) [inline binding/python/multiverso/api.py]: two empty lines according to pep8?

File: binding/python/multiverso/api.py [hunk 2]
```diff
@@ -56,17 +59,20 @@ def get(self):
...
     def add(self, data):
         '''
         Data type of `data` is numpy.ndarray
         '''
-        data = np.array(data)
+        if not isinstance(data, np.ndarray):
+            data = np.array(data)
         assert(data.size == self._size)
-        mv_lib.MV_AddArrayTable(self._handler, (c_float * self._size)(*data.reshape((-1,))), self._size)
+
```
Comment (huxuan) [inline binding/python/multiverso/api.py]: Is this empty line intended? I don't think there need one.

**Comment by huxuan** [comment]
Cool changes, only two tiny style guide concerns. Seems that we are both learning to use theano/torch more elegantly. :-)

File: binding/python/multiverso/api.py (hunk 1 above)
Comment (litterbear) [inline binding/python/multiverso/api.py]: Yes, it conflicts with PEP8
I'm wondering why my pep8 checker didn't report it... You really have beaten it... Lol

File: binding/python/multiverso/api.py (hunk 2 above)
Comment (litterbear) [inline binding/python/multiverso/api.py]: removed now

File: binding/python/multiverso/api.py (hunk 1 above)
Comment (you-n-g) [inline binding/python/multiverso/api.py]: Yes, it conflicts with PEP8
I'm wondering why my pep8 checker didn't report it... You really have beaten it... Lol
 Add a line note

File: binding/python/multiverso/api.py (hunk 2 above)
Comment (you-n-g) [inline binding/python/multiverso/api.py]: Deleted now...

File: binding/python/multiverso/api.py (hunk 1 above)
Comment (huxuan) [inline binding/python/multiverso/api.py]: 强迫症改变世界 ;-)

**Pull request merged**